installer/scripts/telemetry.rb

```diff
@@ -0,0 +1,92 @@
+class Telemetry
+
+    def self.strongtyped_accessor(name, type)
+      # setter
+      self.class_eval("def #{name}=(value);@#{name}=value;end")
+    
+      # getter
+      self.class_eval("def #{name};
+        if @#{name}.is_a? #{type}
+          @#{name}
+        else
+          raise ArgumentError, \"Invalid data type. #{name} should be type #{type}\"
+        end
+      end")
+    end
+
+    strongtyped_accessor :PercentUserTime, Integer
+    strongtyped_accessor :PercentPrivilegedTime, Integer
+    strongtyped_accessor :UsedMemory, Integer
+    strongtyped_accessor :PercentUsedMemory, Integer
+    strongtyped_accessor :OSName, String
+    strongtyped_accessor :OSManufacturer, String
+    strongtyped_accessor :OSVersion, String
+    strongtyped_accessor :InContainer, String
+end
+
+
+def get_telemetry_data(telemetry, os_info, process_stats)
+  if File.exist?('/root/.dockerenv') || File.exist?('/root/.dockerinit')
+    telemetry.InContainer = "True"
+  else
+    telemetry.InContainer = "False"
+  end
+
+  if ENV['TEST_WORKSPACE_ID'].nil? && ENV['TEST_SHARED_KEY'].nil? && File.exist?('/etc/opt/microsoft/omsagent/conf/omsadmin.conf')
+    stats=`/opt/omi/bin/omicli wql root/scx "SELECT PercentUserTime, PercentPrivilegedTime, UsedMemory, PercentUsedMemory FROM SCX_UnixProcessStatisticalInformation where Name like 'omsagent'" | grep =`
+    File.write(process_stats, stats)
+    stats.each_line do |line|
+      telemetry.PercentUserTime = line.sub("PercentUserTime=","").strip.to_i if line =~ /PercentUserTime/
+      telemetry.PercentPrivilegedTime = line.sub("PercentPrivilegedTime=", "").strip.to_i if  line =~ /PercentPrivilegedTime/
+      telemetry.UsedMemory = line.sub("UsedMemory=", "").strip.to_i if line =~ / UsedMemory/
+      telemetry.PercentUsedMemory = line.sub("PercentUsedMemory=", "").strip.to_i if line =~ /PercentUsedMemory/
+    end
+  end
+
+  File.open(os_info).each_line do |line|
+    telemetry.OSName = line.sub("OSName=","").strip if line =~ /OSName/
+    telemetry.OSManufacturer = line.sub("OSManufacturer=","").strip if line =~ /OSManufacturer/
+    telemetry.OSVersion = line.sub("OSVersion=","").strip if line =~ /OSVersion/
+  end
+end
+        
+def to_xml(telemetry, path)
+  File.open(path, 'a') do |f|
```

Comment:
Can you investigate using some library to convert the ruby class into the xml string?